src/hb-serialize.hh

```diff
@@ -117,7 +127,12 @@ struct hb_serialize_context_t
     object_pool.fini ();
   }
 
-  bool in_error () const { return !this->successful; }
+  bool in_error () const { return errors & HB_SERIALIZE_ERR_OTHER; }
```

Comment:
If the check the old code, the only place where `ran_out_of_room` was being set to true, `successful` was also being set to false. So, `in_error` should simply return `bool (errors)`. *Any* error will make `in_error()` fire. The `ERROR_OTHER` simply means "something else".

Reply:
in_error() is currently treated as a check for fatal errors. In my first attempt at this change I did indeed just use bool (errors) but that caused subsetting to fail when offset overflows happened.

Instead, I think I'll make this more clear and rename to fatal_error () and then check if any error flag other than offset overflow is set.